add supported-networks list with firehose protocol info (#91)
streamingfast/firehose-ethereum: SUPPORTED-NETWORKS.md

# Firehose Ethereum Instrumentation

> Firehose provides TWO mechanisms to extract data from an ethereum-like blockchain to `sf.ethereum.type.v2.Block` message:

Reviewer: The `to` is a bit weird at the end

> Firehose provides TWO mechanisms to extract data from an ethereum-like blockchain to `sf.ethereum.type.v2.Block` message:

Reviewer: ```suggestion
Firehose provides **two** mechanisms to extract data from an Ethereum-like blockchain to `sf.ethereum.type.v2.Block` message:
```

I feel the `TWO` is intense.

> 1. Running an instrumented node (data is extracted while the EVM actually executes the transactions) -> produces EXTENDED blocks

Reviewer: ```suggestion
1. Running an instrumented node (data is extracted while the node actually executes the transactions) -> produces `EXTENDED` blocks
```

> 2. Polling an standard node with a series of RPC calls (ex: get_block) -> produces BASE blocks

Reviewer: ```suggestion
1. Polling an standard node with a series of RPC calls (ex: get_block) -> produces BASE blocks
```

It's preferable (at least in my opinion) to always use `1.` and let the template generator generates the numbers.

> 2. Polling an standard node with a series of RPC calls (ex: get_block) -> produces BASE blocks

Reviewer: ```suggestion
2. Polling an standard node with a series of RPC calls (ex: `eth_getBlock`) -> produces BASE blocks
```

The "Extended" vs "Base" distinction appears in the "DetailLevel" field of the Block message 
  ref: https://github.com/streamingfast/firehose-ethereum/blob/develop/proto/sf/ethereum/type/v2/type.proto#L85

## Instrumented nodes

### Description 

> Most instrumented nodes for ethereum-like networks have parts of their codebase coming from geth (https://github.com/ethereum/go-ethereum)

Reviewer: ```suggestion
Most instrumented nodes for Ethereum-like networks have parts of their codebase coming from geth (https://github.com/ethereum/go-ethereum)
```

> Nodes that are based on geth versions prior to [v1.14.0](https://github.com/ethereum/go-ethereum/releases/tag/v1.14.0) contain a lot of firehose-specific instrumentation hooks send each 'event' to the firehose reader which then assembles the block. (firehose protocol < fh3)

Reviewer: ```suggestion
Nodes that are based on geth versions prior to [v1.14.0](https://github.com/ethereum/go-ethereum/releases/tag/v1.14.0) contain a lot of Firehose-specific instrumentation hooks send each 'event' to the firehose reader which then assembles the block. (firehose protocol < `3.0`)
```

> Nodes that are based on geth versions >= v1.14.0 can use the "Live tracing" feature (https://github.com/ethereum/go-ethereum/blob/master/core/tracing/CHANGELOG.md) to send fully-formed blocks to the firehose reader (firehose protocol == fh3)

Reviewer: ```suggestion
Nodes that are based on `geth` versions >= v1.14.0 can use the "Live tracing" feature (https://github.com/ethereum/go-ethereum/blob/master/core/tracing/CHANGELOG.md) to send fully-formed blocks to the firehose reader (firehose protocol == `3.0`)
```

### Supported networks status

| Protocol | Networks                             | Upstream has v1.14 tracing | Firehose protocol version |
| -------- | ------------------------------------ | -------------------------- | ------------------------- |
| Ethereum | Mainnet<br>Sepolia<br>Holesky        | Yes                        | fh2.4                     |

> | Polygon  | Matic (mainnet)<br>Mumbai (testnet)  | Yes (Coming in v1.5.0)     | fh2.4                     | 

Reviewer: ```suggestion
| BSC  | BSC Mainnet<br>BSC Testnet  | No (Patched in SF github )     | fh2.5                     | 
| Polygon  | Matic (mainnet)<br>Mumbai (testnet)  | Yes (Coming in v1.5.0)     | fh2.4                     | 
```

> | Arbitrum | arb-one*<br>sepolia<br>nova          | No (Patched in SF github ) | fh3                       |

Reviewer: Why is `sepolia` in there?

| Optimism | Mainnet**<br>Sepolia<br>Base         | Yes                        | fh3                       |

> 

Reviewer: There is Sei also (for the EVM part).

 * `*` arb-one is not supported by the instrumented node below block 22207818
 * `**` optimism mainnet is not supported by the instrumented node below block 105235064

## RPC-polled

> Most ethereum-like networks can work directly using the RPC poller

Reviewer: Should we list them actually?
